src/components/sandpack-embed/index.tsx

```diff
@@ -1,44 +1,87 @@
+  CodeEditorProps,
+  PreviewProps,
+  SandpackCodeEditor,
+  SandpackLayout,
+  SandpackLayoutProps,
+  SandpackPreview,
+  SandpackProvider,
+} from '@codesandbox/sandpack-react'
+import { nightOwl } from '@codesandbox/sandpack-themes'
 
-type Props = BoxProps &
-  SandpackProps & {
-    dependencies?: Record<string, string>
-    devDependencies?: Record<string, string>
+type Props = BoxProps & {
+  dependencies?: Record<string, string>
+  devDependencies?: Record<string, string>
+  layoutOptions?: SandpackLayoutProps
+  editorOptions?: CodeEditorProps
+  previewOptions?: PreviewProps
+  files: {
+    [x: string]: string
   }
+  isHorizontal?: boolean
+}
 
-const SandpackEmbed = ({ dependencies, devDependencies, ...props }: Props) => (
-  <Box
-    as={Sandpack}
-    {...props}
-    options={{
-      ...props.options,
-      showLineNumbers: true,
-    }}
-    theme='dark'
-    template='react-ts'
-    customSetup={{
-      dependencies: {
-        'react-icons': '3.11.0',
-        '@chakra-ui/react': 'latest',
-        '@chakra-ui/icons': 'latest',
-        '@emotion/react': '^11.7.0',
-        '@emotion/styled': '^11.6.0',
-        'framer-motion': '^4.1.17',
-        react: '^18.0.0',
-        'react-dom': '^18.0.0',
-        'react-scripts': '^4.0.0',
-        ...dependencies,
-      },
-      devDependencies: {
-        '@types/react': '^18.0.0',
-        '@types/react-dom': '^18.0.0',
-        typescript: '^4.0.0',
-        ...devDependencies,
-      },
-    }}
-  />
-)
+const SandpackEmbed = ({
+  dependencies,
+  devDependencies,
+  layoutOptions,
+  editorOptions,
+  previewOptions,
+  files,
+  isHorizontal = false,
+}: Props) => {
+  return (
+    <SandpackProvider
+      files={files}
+      theme={nightOwl}
+      template='react-ts'
+      customSetup={{
+        dependencies: {
+          'react-icons': '3.11.0',
+          '@chakra-ui/react': 'latest',
+          '@chakra-ui/icons': 'latest',
+          '@emotion/react': '^11.7.0',
+          '@emotion/styled': '^11.6.0',
+          'framer-motion': '^4.1.17',
+          react: '^18.0.0',
+          'react-dom': '^18.0.0',
+          'react-scripts': '^4.0.0',
+          ...dependencies,
+        },
+        devDependencies: {
+          '@types/react': '^18.0.0',
+          '@types/react-dom': '^18.0.0',
+          typescript: '^4.0.0',
+          ...devDependencies,
+        },
+      }}
+    >
+      <Box
+        as={SandpackLayout}
+        sx={{
+          '--sp-layout-height': 'auto',
+        }}
+        style={{ flexDirection: isHorizontal ? 'row' : 'column-reverse' }}
+        {...layoutOptions}
+      >
+        <SandpackCodeEditor
+          showLineNumbers
+          style={{
+            maxHeight: isHorizontal ? '600px' : '500px',
+            minWidth: '400px',
+          }}
+          {...editorOptions}
+        />
+        <Box
+          as={SandpackPreview}
+          minHeight='350px'
+          sx={{ '& iframe': { flex: 'initial', flexGrow: 1 } }}
```

Comment:
@noobinthisgame

This allows the content in the preview to stretch the height of the container so the scroll never triggers.

Wasn't sure if this is a necessary adjustment, so I added it in case you wanted to check it out.